runtime/lua/vim/shared.lua

```diff
@@ -104,13 +104,13 @@ end
 ---
 ---@see |vim.gsplit()|
 ---
----@param s String to split
----@param sep Separator string or pattern
----@param kwargs Keyword arguments:
+---@param s string String to split
+---@param sep string Separator string or pattern
+---@param kwargs table|boolean Keyword arguments:
```

Comment:
```suggestion
---@param kwargs table Keyword arguments:
```

`boolean` is only supported for BWC. I wouldn't keep promoting that in the docs. 
Otherwise we'd also have to document the behavior of the boolean parameter ( that it sets `plain`)